src/darwin/Framework/CHIP/templates/MTRClusterNames-src.zapt

```diff
@@ -0,0 +1,88 @@
+{{> header excludeZapComment=true}}
+
+
+#import <Foundation/Foundation.h>
+
+#import "MTRClusterNames.h"
+
+// @implementation MTRClusterAttributeIDType
+
+#pragma mark - Clusters IDs
+
+NSString * MTRClusterNameForID(MTRClusterIDType clusterID)
+{
+    NSString * result = nil;
+    
+    switch (clusterID) {
+
+{{#zcl_clusters}}
+{{#if (isSupported (asUpperCamelCase label preserveAcronyms=true) isForIds=true)}}
+{{~#*inline "cluster"}}{{asUpperCamelCase label preserveAcronyms=true}}{{/inline~}}
+        case MTRClusterIDType{{>cluster}}ID:
+            result = @"{{>cluster}}";
+            break;
+{{/if}}
+{{/zcl_clusters}}
+
+        default:
+            result = [NSString stringWithFormat:@"<Unknown clusterID %d>", clusterID];
+    }
+    
+    return result;
+}
+
+
+#pragma mark - Attributes IDs
+
+NSString * MTRAttributeNameForID(MTRClusterIDType clusterID, MTRAttributeIDType attributeID)
+{
+    NSString * result = nil;
+
+    switch (clusterID) {
+
+{{#zcl_clusters}}
+{{#if (isSupported (asUpperCamelCase label preserveAcronyms=true) isForIds=true)}}
+{{~#*inline "cluster"}}{{asUpperCamelCase label preserveAcronyms=true}}{{/inline~}}
+        case MTRClusterIDType{{>cluster}}ID:
+
+            switch (attributeID) {
+
+{{/if}}
+
+{{#*inline "attributeIDs"}}
+{{#zcl_attributes_server}}
+{{~#*inline "cluster"}}{{asUpperCamelCase ../clusterName preserveAcronyms=true}}{{/inline~}}
+{{~#*inline "attribute"}}{{asUpperCamelCase label preserveAcronyms=true}}{{/inline~}}
+{{#first}}
+{{#if (isSupported (asUpperCamelCase ../clusterName preserveAcronyms=true) isForIds=true)}}
```

Comment:
Why is this check needed? We already checked the cluster is supported if we're here, no?

Reply:
This is coming from src/darwin/Framework/CHIP/templates/MTRClusterConstants.zapt where this was handled in a similar way.

The if above has been /if'ed, so I don't think we can still assume the cluster is supported here.

Reply:
Ah, I see; the issue is that if we are here we know the cluster is in fact supported, but we could fail to compile if we output cases for the one that are not supported as ids.  Alright, makes sense.